tools/privacy-guardian.ps1

```
@@ -0,0 +1,99 @@
+    try {
+        $MatchesFound = Select-String -Path $TargetFile -Pattern $Pattern -AllMatches
+        if ($MatchesFound) {
+            Write-Host "`n⚠️  POTENTIAL LEAK DETECTED:" -ForegroundColor $Red
+            foreach ($match in $MatchesFound) {
+                Write-Host "Line $($match.LineNumber): $($match.Line.Trim())" -ForegroundColor $Red
+            }
+            $LeakFound = $true
+        }
+    } catch {
+        # Silently ignore regex errors if any
+    }
```

PSScriptAnalyzer (1.25.0)

[warning] Empty catch block is used. Please use Write-Error or throw statements in catch blocks.

(PSAvoidUsingEmptyCatchBlock)